google/cloud/spanner_v1/_opentelemetry_tracing.py

```diff
@@ -15,19 +15,39 @@
 """Manages OpenTelemetry trace creation and handling"""
 
 from contextlib import contextmanager
+import os
 
 from google.api_core.exceptions import GoogleAPICallError
 from google.cloud.spanner_v1 import SpannerClient
 
 try:
     from opentelemetry import trace
     from opentelemetry.trace.status import Status, StatusCode
+    from opentelemetry.semconv.trace import SpanAttributes
 
     HAS_OPENTELEMETRY_INSTALLED = True
+    DB_SYSTEM = SpanAttributes.DB_SYSTEM
+    DB_NAME = SpanAttributes.DB_NAME
+    DB_CONNECTION_STRING = SpanAttributes.DB_CONNECTION_STRING
+    NET_HOST_NAME = SpanAttributes.NET_HOST_NAME
+    DB_STATEMENT = SpanAttributes.DB_STATEMENT
 except ImportError:
     HAS_OPENTELEMETRY_INSTALLED = False
 
 
+EXTENDED_TRACING_ENABLED = os.environ.get('SPANNER_ENABLE_EXTENDED_TRACING', '') == 'true'
+
+
+def annotate_with_sql_statement(span, sql):
+    """
+    annotate_sql_statement will set the attribute DB_STATEMENT
+    to the sql statement, only if SPANNER_ENABLE_EXTENDED_TRACING=true
+    is set in the environment.
+    """
+    if EXTENDED_TRACING_ENABLED:
+        span.set_attribute(DB_STATEMENT, sql)
+
+
 @contextmanager
 def trace_call(name, session, extra_attributes=None):
     if not HAS_OPENTELEMETRY_INSTALLED or not session:
```

Comment:
1. We are using global tracer provider here by default. But we should also give an option for customer to pass customer tracer provider for spanner traces and fallback to global if not provided.
2. Set the instrumenting_module_name to `cloud.google.com/language`, instrumenting_library_version when calling get_tracer()

Reply:
All done! Thank you @harshachinta!